### ISSUE TITLE
Clarify license and redistributability of bundled WinSDK headers

### ISSUE THREAD

**Issue by mstorsjo** [microsoft/win32metadata#766]
The repo contains a toplevel file LICENSE, indicating that the files are covered by the MIT license (allowing redistribution etc).

The repo also contains a number of headers from WinSDK (under `generation/WinSDK/RecompiledIdlHeaders`). The headers in WinSDK generally aren't redistributable (the WinSDK license doesn't allow redistribution of anything else than specific redistributables). The files here also contain such a license header that don't seem to imply that it's intended to be freely redistributable under the MIT license, e.g. looking at `um/winnt.h`:

```
/*++ BUILD Version: 0072     Increment this if a change has global effects
    
Copyright (c) Microsoft Corporation. All rights reserved.     
```
(Which explicitly says `All rights reserved.` and nothing about the file being redistributable under any OSS license).

Also as noted in #387, the NuGet package form links to the WinSDK license (which doesn't permit redistribution).

Thus, is it possible to make it clearer that all the included files, including the WinSDK headers, now are redistributable according to the MIT license (which seems to be the intent?), in contrast to when receiving those headers from a regular WinSDK installation?

**Comment by danny-laa** [comment]
FYI, MIT License includes an `All rights reserved.` clause too.

**Comment by mstorsjo** [comment]
> FYI, MIT License includes an `All rights reserved.` clause too.

Hmm, at least my reading of it, at https://opensource.org/licenses/MIT, doesn't contain that phrase in the license itself. But anyway, even if the phrase is used... If it's directly followed by the text granting specific permissions, it's a different thing that when the clause stands on its own with no further grants or even references to other license files.

**Comment by riverar** [comment]
@mikebattista Bubbling this up as it has become quite popular.

**Comment by mikebattista** [comment]
We haven't changed the license of the original SDK headers. The license in this repo is for the metadata and tooling around it.

**Comment by g2p** [comment]
This repo contains foundational metadata for various language bindings. The licence files here are clear enough about the licence for the tools, but the metadata is derived from non redistributable headers, and it is not obvious that you relicense it.

**Comment by slimsag** [comment]
> We haven't changed the license of the original SDK headers.

Is this to be interpreted as "the files under generation/WinSDK/RecompiledIdlHeaders in this repository are not open source / MIT licensed"?

**Comment by slimsag** [comment]
@mikebattista @riverar just curious if you could clarify the above still

**Comment by ChrisDenton** [comment]
So `win32metadata/generation/WinSDK/RecompiledIdlHeaders/` are not open source?

Are `Windows.Win32.winmd` and `Windows.Win32.Interop.dll` MIT licensed, despite the fact they are derived from headers with a restrictive license?

**Comment by ChrisDenton** [comment]
> You can get the same files at https://github.com/microsoft/win32metadata/tree/main/scripts/BaselineWinmd to workaround that until it's changed.

It would be nice if this was made explicit somewhere so people know it's safe to create works derived from them even though they are themselves derived from sources that have a restrictive license. This hasn't been clear to people, IMHO.

Actually, it'd be great if your "general guidance" were in the readme or a FAQ to avoid this confusion in the future.

**Comment by riverar** [comment]
That's fair, I can prepare a PR to the README.

> It would be nice if this was made explicit somewhere so people know it's safe to create works derived from them even though they are themselves derived from sources that have a restrictive license.

There is no concept of "derived from sources that have a restrictive license" in play here, thankfully! Microsoft owns the source material and can do whatever they want and license any number of derivatives with any license they choose.

**Comment by ChrisDenton** [comment]
Right, I'm agreeing with that. But that's not necessarily clear they are being relicensed under MIT if you're not a lawyer and haven't read through this thread.

Some parts of this repo are MIT licensed, some parts aren't and those particular files are binary output from the generator so there's room for doubt as to if the repo license applies to them. The NuGet situation only confuses things more.

**Comment by riverar** [comment]
Got it, yep. On the same page now.
